typescript/sdk/src/deploy/verify/utils.ts

```diff
@@ -84,3 +91,204 @@ export function shouldAddVerificationInput(
       existingArtifact.isProxy === artifact.isProxy,
   );
 }
+
+/**
+ * Retrieves the constructor args using their respective Explorer and/or RPC (eth_getTransactionByHash)
+ */
+export async function getConstructorArgumentsApi({
+  blockExplorerApiKey,
+  blockExplorerApiUrl,
+  chainName,
+  contractAddress,
+  bytecode,
+  multiProvider,
+}: {
+  blockExplorerApiKey: string | undefined;
+  blockExplorerApiUrl: string;
+  bytecode: string;
+  chainName: string;
+  contractAddress: string;
+  multiProvider: MultiProvider;
+}): Promise<string> {
+  const { family } = multiProvider.getExplorerApi(chainName);
+
+  let constructorArgs: string;
+  switch (family) {
+    case ExplorerFamily.Routescan:
+    case ExplorerFamily.Etherscan:
+      constructorArgs = await getEtherscanConstructorArgs({
+        blockExplorerApiKey,
+        blockExplorerApiUrl,
+        chainName,
+        contractAddress,
+        bytecode,
+        multiProvider,
+      });
+      break;
+    case ExplorerFamily.Blockscout:
+      constructorArgs = await getBlockScoutConstructorArgs({
+        blockExplorerApiUrl,
+        contractAddress,
+      });
+      break;
+    default:
+      throw new Error(`Explorer Family ${family} unsupported`);
+  }
+
+  return constructorArgs;
+}
+
+export async function getEtherscanConstructorArgs({
+  blockExplorerApiKey,
+  blockExplorerApiUrl,
+  bytecode,
+  chainName,
+  contractAddress,
+  multiProvider,
+}: {
```

Comment:
It seems that these 2 functions have the same  input `type`, maybe it would be a good idea to create a single type def and reuse it for both

Reply:
Generally yes! 
but i've reduced the parameters here by removing the blockExplorer keys and using the registry now, so maybe not worth it to add a type. lmk if you feel like i should :)

Reply:
you can keep it like this np! Generally, I like to create "base types" for types that have common fields and share a common context but I'll leave it up to you is not a deal breaker